File: mobile/lib/graphql/auth_operations.dart
import 'package:graphql_flutter/graphql_flutter.dart';

/// Documents GraphQL pour l'authentification.
///
/// `gql()` parse une chaîne GraphQL en `DocumentNode`, le format attendu
/// par `MutationOptions(document: ...)` / `QueryOptions(document: ...)`.

/// Mutation `login` : authentifie un utilisateur (parent ou enfant).
/// Retourne un `AuthPayload` contenant le JWT (`token`) et l'utilisateur.
final kLoginMutation = gql(r'''
  mutation Login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        id
        email
        role
        firstName
        lastName
      }
    }
  }
''');

/// Query `me` : retourne l'utilisateur courant (authentifié via JWT).
/// Sert à vérifier que le token est valide et récupérer le profil.
final kMeQuery = gql(r'''
  query Me {
    me {
      id
      email
      role
      firstName
      lastName
    }
  }
''');

File: web/src/lib/auth-operations.ts
// Documents GraphQL partagés pour l'authentification (côté serveur et client).
//
// Centralisés ici pour éviter la duplication entre les Route Handlers
// (server-side, via fetch) et les composants client (via Apollo Client).

export const LOGIN_MUTATION = /* GraphQL */ `
  mutation Login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        id
        email
        role
        firstName
        lastName
      }
    }
  }
`;

export const SIGNUP_MUTATION = /* GraphQL */ `
  mutation Signup(
    $firstName: String!
    $lastName: String!
    $email: String!
    $password: String!
    $familyName: String!
  ) {
    signup(
      firstName: $firstName
      lastName: $lastName
      email: $email
      password: $password
      familyName: $familyName
    ) {
      token
      user {
        id
        email
        role
        firstName
        lastName
      }
    }
  }
`;

export const ME_QUERY = /* GraphQL */ `
  query Me {
    me {
      id
      email
      role
      firstName
      lastName
    }
  }
`;

// Type du profil utilisateur retourné par `me` / `login` / `signup`.
export type AppUser = {
  id: string;
  email: string;
  role: 'PARENT' | 'CHILD';
  firstName: string;
  lastName: string;
};
